code/go/0chain.net/chaincore/block/entity.go

```diff
@@ -905,9 +905,7 @@ func (b *Block) ComputeState(ctx context.Context, c Chainer) error {
 	}
 
 	if len(b.Events) > 0 && c.GetEventDb() != nil {
-		go func() {
-			c.GetEventDb().AddEvents(ctx, b.Events)
-		}()
+		c.GetEventDb().AddEvents(ctx, b.Events)
```

Comment:
Since AddEvents is now returning an error we should log it.

Reply:
why is it done in blocking manner now without goroutine?

Reply:
errors logged. 
>why is it done in a blocking manner now without goroutine?

We are actually using the buffered channel. Which if full could block I agree but we have a buffer of 10000 as of now. Are the events going to exceed that in a short time?